src/test/kotlin/sirius/kernel/commons/StringsTest.kt

```diff
@@ -332,4 +332,90 @@ class StringsTest {
         assertEquals("", Strings.limit("ABCDEFGHIJKLMNOP", -1000, false))
     }
 
+    @Test
+    fun truncatedMiddle() {
+        // Szenario 1: no input present
+        assertEquals("", Strings.truncateMiddle(null, 10, 10))
+        assertEquals("", Strings.truncateMiddle(null, 1, 10))
+        assertEquals("", Strings.truncateMiddle(null, 10, 1))
+        assertEquals("", Strings.truncateMiddle(null, 0, 1))
+        assertEquals("", Strings.truncateMiddle(null, 1, 0))
+        assertEquals("", Strings.truncateMiddle(null, 1, -1))
+        assertEquals("", Strings.truncateMiddle(null, -1, 0))
+        assertEquals("", Strings.truncateMiddle(null, -1, -1))
+        assertEquals("", Strings.truncateMiddle(null, 0, 0))
+        // Szenario 2: input present but empty
+        assertEquals("", Strings.truncateMiddle("", 10, 10))
+        assertEquals("", Strings.truncateMiddle("", 1, 10))
+        assertEquals("", Strings.truncateMiddle("", 10, 1))
+        assertEquals("", Strings.truncateMiddle("", 0, 1))
+        assertEquals("", Strings.truncateMiddle("", 1, 0))
+        assertEquals("", Strings.truncateMiddle("", 1, -1))
+        assertEquals("", Strings.truncateMiddle("", -1, 0))
+        assertEquals("", Strings.truncateMiddle("", -1, -1))
+        assertEquals("", Strings.truncateMiddle("", 0, 0))
+        // Szenario 3: input present and length is greater than input.length() and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, 0))
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 10, -10))
+        // Szenario 4: input present and length is smaller than input.length() and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG", 10, 0))
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG", 10, -10))
+        assertEquals("ABCDEFG-AB…[truncated]", Strings.truncateMiddle("ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG-ABCDEFG", 10, -10))
+        // Szenario 5: input present and length is equal to input.length() and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, 0))
+        assertEquals("ABCDEFG", Strings.truncateMiddle("ABCDEFG", 7, -10))
+        // Szenario 6: input present and length is 0 and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", 0, 0))
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", 0, -10))
+        // Szenario 7: input present and length is negative and charactersToPreserveAtTheEnd is 0 or negative
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", -10, 0))
+        assertEquals("…[truncated]", Strings.truncateMiddle("ABCDEFG", -10, -10))
```

Comment:
In my opinion these cases are bad behaviour, as the replacement is longer than the input string

Reply:
the behavior is copied from the old `limit`. 
If I change it, it gets changed there as well. 
If you are oky with that I can change it.